backup-s3/src/main/scala/io/aiven/guardian/kafka/backup/s3/BackupClient.scala

```diff
@@ -200,30 +199,36 @@ class BackupClient[T <: KafkaConsumerInterface](maybeS3Settings: Option[S3Settin
       }
 
   private[s3] def kafkaBatchSink
-      : Sink[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext]), NotUsed] =
+      : Sink[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext]), Future[Done]] =
     // See https://pekko.apache.org/docs/pekko/current/stream/operators/Partition.html for an explanation on Partition
-    Sink.fromGraph(GraphDSL.create() { implicit builder =>
-      import GraphDSL.Implicits._
-      val partition = builder.add(
-        new Partition[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext])](
-          outputPorts = 2,
-          {
-            case (_: SuccessfulUploadPart, _) => 0
-            case (_: FailedUploadPart, _)     => 1
-          },
-          eagerCancel = true
-        )
-      )
-      partition.out(0) ~> successSink
-        .contramap[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext])] {
+    Sink.fromGraph(
+      GraphDSL.createGraph(
+        successSink.contramap[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext])] {
           case (response, value) => (response.asInstanceOf[SuccessfulUploadPart], value)
-        }
-      partition.out(1) ~> failureSink
-        .contramap[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext])] {
+        },
+        failureSink.contramap[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext])] {
           case (response, value) => (response.asInstanceOf[FailedUploadPart], value)
         }
-      SinkShape(partition.in)
-    })
+      ) { (successSinkMat, failureSinkMat) =>
```

Comment:
Since we are passing in the sinks explicitly now, we have to define how to combine the materialization of the sinks. In Pekko terminology, materialization is the result value of an activation of a sink (i.e. for a Kafka publisher sink, whenever the we push a value into the Kafka publisher then the materialized value is a result of that action which with the Kafka example would typically be a `Future[NotUsed]` or a `Future[Done]`

For this specific case of `successSinkMat`/`failureSinkMat`, they are both `Future`'s that contain values we don't really care about. The only thing we do care about is if at least one of the `Future'`s have failed. Also we don't care about ordering and we can wait on the value concurrently, hence why we use `Future.sequence` which will wait for both results asynchronously